crates/test/mock-consensus/src/lib.rs

```diff
@@ -1,45 +1,19 @@
-pub struct TestNode<S> {
-    #[allow(dead_code)]
-    abci_server: S,
-    _last_app_hash: Vec<u8>,
+//! `penumbra-mock-consensus` is a library for testing consensus-driven applications.
+//
+//  see penumbra-zone/penumbra#3588.
+
+mod block;
+mod builder;
+
+// TODO(kate): this is a temporary allowance while we set the test node up.
+#[allow(dead_code)]
+pub struct TestNode<C> {
+    consensus: C,
+    last_app_hash: Vec<u8>,
 }
 
-pub mod block {
-    use crate::TestNode;
-
-    struct _Builder<'e, C> {
-        engine: &'e mut TestNode<C>,
-    }
-}
-
-pub struct Builder;
-
-impl<A> TestNode<A> {
-    pub fn builder() -> Builder {
-        Builder
-    }
-}
-
-impl Builder {
-    // TODO: add other convenience methods for validator config?
-
-    /// Creates a single validator with a randomly generated key.
-    pub fn single_validator(self) -> Self {
+impl<C> TestNode<C> {
+    pub fn next_block() -> tendermint::Block {
```

Comment:
i think this could probably not be a `pub fn`, i'm not sure when we'll need to access the block itself (?)

Reply:
137eb1719